### PR TITLE
fix(backend): critical bug

### LINKED ISSUE #2432: 🐛 [Bug] Critical Bug in current transaction implementation
## 🐛 Bugreport

Critical Bug in current transaction implementation

- [ ] Fix it

### PR DESCRIPTION
**Pull request by Mogge**
### Issues
- fixes #2432 

### Todo

- [X] None

### CODE REVIEW

File: backend/src/graphql/resolver/TransactionResolver.ts [hunk 1]
```diff
@@ -324,22 +328,7 @@ export class TransactionResolver {
...
-    */
-    // const recipientUser = await dbUser.findOne({ id: emailContact.userId })
-
-    /* Code inside this if statement is unreachable (useless by so), 
-    in findUserByEmail() an error is already thrown if the user is not found
-    */
-    if (!recipientUser) {
-      logger.error(`unknown recipient to UserContact: email=${email}`)
-      throw new Error('unknown recipient')
-    }
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/TransactionResolver.ts:342]: ?

File: backend/src/graphql/resolver/TransactionResolver.ts [hunk 2]
```diff
@@ -324,22 +328,7 @@ export class TransactionResolver {
...
+
     if (recipientUser.deletedAt) {
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/TransactionResolver.ts:332]: does `findUserByEmail` even return `withDeleted`?

Comment (Mogge) [reply backend/src/graphql/resolver/TransactionResolver.ts:332]: Yes, I checked this

File: backend/src/graphql/resolver/TransactionResolver.ts (hunk 1 above)
Comment (Mogge) [reply backend/src/graphql/resolver/TransactionResolver.ts:342]: Code is never reached. `findUserByEmail` throws if user cannot be found

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**